### PR TITLE
refactor(cdk): support noUncheckedIndexedAccess

### PR DESCRIPTION
**Pull request by splincode**

### CODE REVIEW

File: projects/cdk/directives/pan/pan.service.ts [hunk 1]
```diff
@@ -38,8 +38,8 @@ export class TuiPanService extends Observable<readonly [number, number]> {
                     ),
                     pairwise(),
                     map(([first, second]) => {
```
Comment (waterplea) [inline projects/cdk/directives/pan/pan.service.ts:40]: It's pairwise, I thought it is typed as tuple with both element always there? We shouldn't be checking such cases.

File: projects/cdk/directives/swipe/swipe.service.ts [hunk 2]
```diff
@@ -25,14 +25,14 @@ export class TuiSwipeService extends Observable<TuiSwipeEvent> {
                     filter(
                         ([first, second]) =>
                             !!first.touches.length &&
-                            first.touches[0].identifier ===
-                                second.changedTouches[0].identifier,
+                            first.touches.item(0)?.identifier ===
```
Comment (waterplea) [inline projects/cdk/directives/swipe/swipe.service.ts:28]: What is the benefit of using `.item()` over index access?

File: projects/cdk/directives/zoom/zoom.service.ts [hunk 3]
```diff
@@ -41,12 +41,12 @@ export class TuiZoomService extends Observable<TuiZoomEvent> {
                             ),
                             map(({event, delta}) => {
                                 const clientX =
-                                    (event.touches[0].clientX +
-                                        event.touches[1].clientX) /
+                                    ((event.touches.item(0)?.clientX ?? 0) +
+                                        (event.touches.item(0)?.clientX ?? 0)) /
```
Comment (waterplea) [inline projects/cdk/directives/zoom/zoom.service.ts:45]: Please be more attentive, second one should be using 1, not 0

File: projects/cdk/utils/color/hex-to-rgba.ts [hunk 4]
```diff
@@ -39,5 +39,5 @@ export function tuiParseHex(
     const [r, g, b, a] = hexArr?.map(convertHexUnitTo256) ?? [];
     const floatAlpha = getAlphaFloat(a, alpha);
 
-    return [r, g, b, floatAlpha];
+    return [r || NaN, g || NaN, b || NaN, floatAlpha];
```
Comment (waterplea) [inline projects/cdk/utils/color/hex-to-rgba.ts:42]: Let's add default values on line 39 when destructuring

File: projects/cdk/utils/color/rgba-to-hex.ts [hunk 5]
```diff
@@ -3,11 +3,17 @@ export function tuiRgbaToHex(color: string): string {
         throw new Error('Invalid RGBa');
     }
 
-    const rgb: number[] =
+    const rgb =
         (color
             .replaceAll(/\s/g, '')
-            .match(/^rgba?\((\d+),(\d+),(\d+),?([^,\s)]+)?/i) as unknown as number[]) ??
-        [];
+            .match(/^rgba?\((\d+),(\d+),(\d+),?([^,\s)]+)?/i) as unknown as [
```
Comment (waterplea) [inline projects/cdk/utils/color/rgba-to-hex.ts:9]: Are we sure there's exactly this many items in the array here? Maybe it's better to assume some of them might not be there?

File: projects/cdk/directives/pan/pan.service.ts (hunk 1 above)
Comment (splincode) [reply projects/cdk/directives/pan/pan.service.ts:40]: image

`touches[0]` return undefined

Comment (waterplea) [reply projects/cdk/directives/pan/pan.service.ts:40]: Yes, but not `first` or `second`? I mean we do not need `?.` after those.

File: projects/cdk/directives/zoom/zoom.service.ts (hunk 3 above)
Comment (splincode) [reply projects/cdk/directives/zoom/zoom.service.ts]: fixed

File: projects/cdk/utils/color/hex-to-rgba.ts (hunk 4 above)
Comment (splincode) [reply projects/cdk/utils/color/hex-to-rgba.ts]: fixed

File: projects/cdk/directives/pan/pan.service.ts (hunk 1 above)
Comment (splincode) [reply projects/cdk/directives/pan/pan.service.ts:40]: image

Comment (splincode) [reply projects/cdk/directives/pan/pan.service.ts:40]: image

File: projects/cdk/directives/swipe/swipe.service.ts (hunk 2 above)
Comment (splincode) [reply projects/cdk/directives/swipe/swipe.service.ts]: fixed

File: projects/cdk/utils/color/rgba-to-hex.ts (hunk 5 above)
Comment (splincode) [reply projects/cdk/utils/color/rgba-to-hex.ts]: fixed

**Comment by splincode** [comment]
@waterplea please, review

**Review by waterplea — APPROVED** [review]

File: projects/cdk/directives/pan/pan.service.ts [hunk 6]
```diff
@@ -38,8 +38,8 @@ export class TuiPanService extends Observable<readonly [number, number]> {
...
-                        const deltaX = second.clientX - first.clientX;
-                        const deltaY = second.clientY - first.clientY;
+                        const deltaX = (second?.clientX || 0) - (first?.clientX || 0);
+                        const deltaY = (second?.clientY || 0) - (first?.clientY ?? 0);
```
Comment (vladimirpotekhin) [inline projects/cdk/directives/pan/pan.service.ts:42]: nit: || 0

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**